etc/dbus-serialbattery/utils.py

```diff
@@ -58,7 +58,7 @@ def read_serial_data(command, port, baud, length_pos, length_check, length_fixed
             count = 0
             data = bytearray(res)
             while len(data) <= length + length_check:
-                res = ser.read(length + length_check)
+                res = ser.read((length + length_check) - len(data))
```

Comment:
Can you give me more information on why this change was needed as it could influent the existing BMS that use this function. If you give it more bytes to read it, read will still only read what is available in the buffer. So more is good. 
Did you get any problems in the Sinowealth based Daly using the read without the  -len(data) ?